src/kOS/Module/kOSVesselModule.cs

```diff
@@ -350,13 +392,26 @@ private void CheckRehookAutopilot()
         /// <param name="c"></param>
         private void UpdateAutopilot(FlightCtrlState c)
         {
-            if (childParts.Count > 0)
+            if (vessel != null)
             {
-                foreach (var parameter in flightControlParameters.Values)
+                if (childParts.Count > 0)
                 {
-                    if (parameter.Enabled && parameter.IsAutopilot)
+                    foreach (var parameter in flightControlParameters.Values)
                     {
-                        parameter.UpdateAutopilot(c);
```

Comment:
The above is the typical thing that happens in git diff output where a diff looks a lot more complex than it really is when you wrap existing code in a new indented section (in this case all I really did is wrap everything in the ``if (vessel != null)`` check).